pkg/controller/operator.go

```diff
@@ -89,8 +89,10 @@ func (c *Controller) fetchUserCredentials(ctx context.Context, tenant *miniov2.T
 	return userCredentials
 }
 
-func (c *Controller) getTransport() *http.Transport {
-	if c.transport != nil {
+// getTransport returns a *http.Transport with the collection of the trusted CA certificates
+// reload=true forces to reload the CA certificates instead of return the cached transport
+func (c *Controller) getTransport(reload bool) *http.Transport {
```

Comment:
how do we know when we should use `reload`  flag? It would do the reload regardles of the flag if c.transport is null.
I'd propose instead of using a boolean, to differentiate behaviors, to write two separate functions:

`fetchTransportCertificates` (name can be different) and `getTransport:`
```
func (c *Controller) getTransport(reload bool) *http.Transport {
  if c.transport != nil {
		  return c.transport
	  }
 return c.fetchTransportCertificates()
```

Where you want to reload just use `c.fetchTransportCertificates()` and so forth. It would make it a bit more clear cause reading `false` or `true` on the function is a bit confusing (I need to go to the definition to understand what it does). Plus there wouldn't be a need for adding those extra comments of `Reload certificates`.

Reply:
sure, applied suggestion